demos/dqx_llm_demo.py

```diff
@@ -0,0 +1,316 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Using DQX for LLM-based Primary Key Detection
+# MAGIC DQX provides optional LLM-based primary key detection capabilities that can intelligently identify primary key columns from table schema and metadata. This feature uses Large Language Models to analyze table structures and suggest potential primary keys, enhancing the data profiling and quality rules generation process.
+# MAGIC
+# MAGIC The LLM-based primary key detection is completely optional and only activates when users explicitly request it. Regular DQX functionality works without any LLM dependencies.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Install DQX with LLM extras
+# MAGIC
+# MAGIC To enable LLM-based primary key detection, DQX has to be installed with `llm` extras: 
+# MAGIC
+# MAGIC `%pip install databricks-labs-dqx[llm]`
+
+# COMMAND ----------
+
+dbutils.widgets.text("test_library_ref", "", "Test Library Ref")
+
+if dbutils.widgets.get("test_library_ref") != "":
+    %pip install 'databricks-labs-dqx[llm] @ {dbutils.widgets.get("test_library_ref")}'
+else:
+    %pip install databricks-labs-dqx[llm]
+
+# COMMAND ----------
+
+dbutils.library.restartPython()
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.labs.dqx.config import ProfilerConfig, LLMConfig
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Regular Profiling (No LLM Dependencies Required)
+# MAGIC 
+# MAGIC By default, DQX works without any LLM dependencies. Regular profiling functionality is always available.
+
+# COMMAND ----------
+
+# Default configuration - no LLM features
+config = ProfilerConfig()
+print(f"LLM PK Detection: {config.llm_config.enable_pk_detection}")  # False by default
+
+# This works without any LLM dependencies!
+print("✅ Regular profiling works out of the box!")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## LLM-Based Primary Key Detection
+# MAGIC 
+# MAGIC When explicitly requested, DQX can use LLM-based analysis to detect potential primary keys in your tables.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Method 1: Configuration-based enablement
```

Comment:
Remove Method 1. This is a demo, not a documentation, either we showcase usage of workflow here or remove this section. Instead we can explain above that this can be enabled for workflows and programmatic methods.